src/lib/comms/include/sol-connman.h

```diff
@@ -0,0 +1,382 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sol-network.h>
+#include <sol-buffer.h>
+#include <sol-vector.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct sol_connman_service
+ *
+ * @brief service struct
+ */
+struct sol_connman_service;
+
+/**
+ * @struct sol_connman_network_params
+ *
+ * @brief network params
+ *
+ * This struct contains the information of a network.
+ * it has the addr of network link addr, the network of
+ * netmask and its gateway of network.
+ */
+struct sol_connman_network_params {
+    /**
+     * @brief The network devices address
+     */
+    struct sol_network_link_addr addr;
+    /**
+     * @brief The network devices netmask
+     */
+    struct sol_network_link_addr netmask;
+    /**
+     * @brief The network gateway
+     */
+    struct sol_network_link_addr gateway;
+};
+
+/**
+ * @enum sol_connman_service_state
+ *
+ * @brief service state
+ *
+ * One of these must be chosen to set the state of service,
+ * with sol_connman_service_get_state()
+ *
+ * @note SOL_CONNMAN_SERVICE_STATE_REMOVE is used to show
+ * the service has been removed. When it is notified,
+ * the service has been removed in the system.
+ */
+enum sol_connman_service_state {
+    SOL_CONNMAN_SERVICE_STATE_UNKNOWN = 0,
+    SOL_CONNMAN_SERVICE_STATE_IDLE,
+    SOL_CONNMAN_SERVICE_STATE_ASSOCIATION,
+    SOL_CONNMAN_SERVICE_STATE_CONFIGURATION,
+    SOL_CONNMAN_SERVICE_STATE_READY,
+    SOL_CONNMAN_SERVICE_STATE_ONLINE,
+    SOL_CONNMAN_SERVICE_STATE_DISCONNECT,
+    SOL_CONNMAN_SERVICE_STATE_FAILURE,
+    SOL_CONNMAN_SERVICE_STATE_REMOVE,
+};
+
+#define SOL_CONNMAN_SERVICE_TYPE_ETHERNET   "ethernet"  /**< @brief ethernet service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_WIFI       "wifi"      /**< @brief wifi service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_BLUETOOTH  "bluetooth" /**< @brief bluetooth service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_CELLULAR   "cellular"   /**< @brief bluetooth service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_GPS        "gps"       /**< @brief gps service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_VPN        "vpn"       /**< @brief vpn service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_GADGET     "gadget"    /**< @brief gadget service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_P2P        "p2p"       /**< @brief p2p service type string */
+
+/**
+ * @enum sol_connman_state
+ *
+ * @brief the global connection state of system
+ *
+ * One of these must be chosen for the global connection state,
+ * with sol_connman_get_state()
+ *
+ */
+enum sol_connman_state {
+    SOL_CONNMAN_STATE_UNKNOWN   = 0,
+    SOL_CONNMAN_STATE_IDLE,
+    SOL_CONNMAN_STATE_READY,
+    SOL_CONNMAN_STATE_ONLINE,
+    SOL_CONNMAN_STATE_OFFLINE,
+};
```

Comment:
same as above, document each member. Like what is the difference from READY and ONLINE.